### PR TITLE
Fixed #4229

### LINKED ISSUE #4229: The model: `gpt-4` does not exist / fixing defaults in llm_utils.py
### Duplicates

- [X] I have searched the existing issues

### Summary 💡

the defaults need to be adapted for people without access to GPT4 - or the corresponding functions won't work.
The default model for people without GPT4 access should be cfg.fast_llm_model and not cfg.smart_llm_model

It would probably make sense to check once during startup what model is available and then set up the default accordingly.

### Examples 🌈

_No response_

### Motivation 🔦

_No response_

### PR DESCRIPTION
**Pull request by konraddroeske**
Resolves: https://github.com/Significant-Gravitas/Auto-GPT/issues/4229

### Background
The defaults need to be adapted for people without access to GPT4 - or the corresponding functions won't work. The default model for people without GPT4 access should be cfg.fast_llm_model and not cfg.smart_llm_model.

### Changes
Added check_model fn in configurator that sends a request to the OpenAI API using create_chat_completion with the model provided. If an InvalidRequestError is raised, it returns a hard-coded value of "gpt-3.5-turbo".

### Documentation
Comments added to the function and test.

### Test Plan
Checks if fn returns original model argument if no error.
Checks if fn returns "gpt-3.5-turbo" when InvalidRequestError is raised.

### PR Quality Checklist
- [x] My pull request is atomic and focuses on a single change.
- [x] I have thoroughly tested my changes with multiple different prompts.
- [x] I have considered potential risks and mitigations for my changes.
- [x] I have documented my changes clearly and comprehensively.
- [x] I have not snuck in any "extra" small tweaks changes. 
- [x] I have run `black .` and `isort .` against my code to ensure it passes our linter.

### CODE REVIEW

File: autogpt/llm/llm_utils.py [hunk 1]
```diff
@@ -293,3 +293,24 @@ def create_embedding(
...
+            model=cfg.smart_llm_model, messages=messages, temperature=0
+        )
+    except InvalidRequestError:
+        logger.typewriter_log(
+            "WARNING: ",
+            Fore.YELLOW,
+            f"You do not have access to {cfg.smart_llm_model}. Setting default "
+            f"to {cfg.fast_llm_model}.",
+        )
+        cfg.smart_llm_model = cfg.fast_llm_model
```
Comment (k-boikov) [inline autogpt/llm/llm_utils.py]: What if they have set for both fast and smart GPT4?

**Review by k-boikov — CHANGES_REQUESTED** [review]

File: autogpt/llm/llm_utils.py (hunk 1 above)
Comment (konraddroeske) [reply autogpt/llm/llm_utils.py]: I've moved the function into the configurator - if both flags are added and chat gpt check fails, it updates gpt3only to True and gpt4only to False. There's also the same log warning.

Comment (konraddroeske) [reply autogpt/llm/llm_utils.py]: Hey @k-boikov, so I've updated the approach to checking both models individually, and updating to "gpt-3.5-turbo" if either them raises an exception. Is this what you had in mind?

Comment (konraddroeske) [reply autogpt/llm/llm_utils.py]: Hey @k-boikov, changes made:

- check_model() fn added which takes model name and model type, returns original model name if valid, or hard-coded "gpt-3.5-turbo" if not valid
- Models are retrieved from get_models() in APIManager, which uses openai.Model.list()
- Checks for both fast_llm_model and smart_llm_model are done at top of create_config()

Let me know if you'd like any other updates or another approach.

**Review by k-boikov — APPROVED** [review]

**Review by k-boikov — APPROVED** [review]

**Pull request merged**